tests/rptest/tests/acls_test.py

```diff
@@ -282,12 +281,29 @@ def expect_startup_failure(use_tls: bool, authn_method: Optional[str],
 
         self.logger.info(f"startup_should_fail={startup_should_fail}")
 
-        self.prepare_cluster(use_tls,
-                             use_sasl,
-                             enable_authz,
-                             authn_method,
-                             client_auth=client_auth,
-                             expect_fail=startup_should_fail)
+        prepare_failed_auth = False
+        try:
+            self.prepare_cluster(use_tls,
+                                 use_sasl,
+                                 enable_authz,
+                                 authn_method,
+                                 client_auth=client_auth,
+                                 expect_fail=startup_should_fail)
+        except RpkException as e:
+            if "CLUSTER_AUTHORIZATION_FAILED" not in str(e):
+                raise
+            prepare_failed_auth = True
+
+        # these combinations end up causing anonymous user and so we get authz
+        # failed rejected when preparing the cluster above.
+        if authn_method == "none" and enable_authz is True:
+            assert prepare_failed_auth
+        elif authn_method == "none" and enable_authz is None and use_sasl:
+            assert prepare_failed_auth
+        elif authn_method is None and enable_authz is True:
+            assert prepare_failed_auth
+        else:
+            assert not prepare_failed_auth
```

Comment:
This works as a quick fix but I think it masks the problem.

The issue is in in cluster setup (`prepare_cluster()` in this file) where we tell RP to check authz (because `enable_authz=True`), we then create SASL users with ACLs even though we do not supply an authn method. So RP runs authz with the anon user instead of the user with admin ACLs. There is simply a lot of control flow to handle to get that right.